main.go

```diff
@@ -19,86 +19,290 @@ package main
 import (
 	"context"
 	"crypto/rand"
+	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
 	"io"
 	"log"
 	"net"
 	"net/http"
 	"os"
+	"os/signal"
 	"strconv"
+	"strings"
 	"sync"
 	"sync/atomic"
 	"syscall"
 	"time"
 
 	"github.com/dustin/go-humanize"
-	"github.com/google/uuid"
 	"golang.org/x/sys/unix"
 )
 
-var port = func() string {
-	p := os.Getenv("HPERF_PORT")
-	if p == "" {
-		p = "9999"
+func printDataOut(_ context.Context, cancel context.CancelFunc) {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+	go func() {
+		<-sigc
+		httpListener.Close()
+		cancel()
+	}()
+
+	lastPrint := false
+	for {
+		if lastPrint {
+			break
+		}
+		if activeClients.Load() == 0 && activeRequests.Load() == 0 {
+			lastPrint = true
+		}
+		if time.Since(start).Seconds() > float64(seconds) {
+			httpListener.Close()
+			cancel()
+		}
+		time.Sleep(1 * time.Second)
+
+		list := genrateDataPoint()
+
+		if printJSON {
+			for _, v := range list {
+				outb, _ := json.Marshal(v)
+				fmt.Println(string(outb))
+			}
+		} else {
+			printToConsole(list)
+		}
+
+	}
+}
+
+type DataPoint struct {
```

Comment:
```suggestion
type DataPoint struct {
```

Please document the struct